crm_rma_prodlot_invoice/models/stock_transfer_details.py

```diff
@@ -27,38 +27,46 @@ class StockTransferDetails(models.TransientModel):
 
     @api.multi
     def do_detailed_transfer(self):
-        """
-        When incoming type transfer are made and stock move have serial/lot
+        """ When incoming type transfer are made and stock move have serial/lot
         number, the supplier is assigned to the serial/lot number taken from
         picking.
         @return: do_detailed_transfer boolean results
         """
         res = super(StockTransferDetails, self).do_detailed_transfer()
+        self._search_serial_number_to_change_from_transfer()
+        return res
 
-        picking_ids = self.env['stock.picking'].browse(self.picking_id.id)
-        order = picking_ids.sale_id
-        move_ids = order.mapped('picking_ids.move_lines')
-        invoice_lines = order.mapped('invoice_ids.invoice_line')
-
-        if not invoice_lines or not move_ids:
-            return res
-
-        for inv_line in invoice_lines:
-            for mov in move_ids:
+    @api.multi
+    def _search_serial_number_to_change_from_transfer(self):
```

Comment:
Same comment here regarding the ambiguity of the method name. Please make it slightly more obvious from the name of the method that a record is being mutated instead of just searched.